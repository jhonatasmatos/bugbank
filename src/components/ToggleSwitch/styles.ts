import styled, { css } from 'styled-components';

type Props = {
  isChecked: boolean;
};

export const Container = styled.label`
  ${({ theme }) => css`
    border: 1px solid ${theme.colors.primary};
    width: 5rem;
    height: 2rem;
    border-radius: 1rem;
    display: flex;
    align-items: center;
    justify-content: center;
    cursor: pointer;
    position: relative;
    background: ${theme.colors.gray};

    ${(p: Props) =>
      p.isChecked &&
      css`
        background: ${theme.colors.primary};
      `}
  `}
`;

export const Input = styled.input`
  width: 100%;
  height: 100%;
  opacity: 0;
  cursor: pointer;
`;

export const Span = styled.span`
  ${({ theme }) => css`
    position: absolute;
    width: 2rem;
    height: 2rem;
    border-radius: 1rem;
    background: ${theme.colors.white};
    left: 0;
    cursor: pointer;
    box-shadow: 2px 2px 4px ${theme.colors.primary};
    transition: all 0.3s;

    ${(p: Props) =>
      p.isChecked &&
      css`
        left: 60%;
      `}
  `}
`;
