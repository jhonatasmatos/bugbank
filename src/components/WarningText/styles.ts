import styled, { css } from 'styled-components';

interface IWarnigWrapper {
  color?: 'white' | 'secondary';
}

export const WarnigWrapper = styled.p`
  ${({ theme }) => css`
    padding: 1.6rem;
    border: 1px solid ${theme.colors.secondary};
    border-radius: 0.8rem;
    color: ${theme.colors.secondary};
    font-style: italic;
    font-size: 1.3rem;
    text-align: center;
    line-height: 1.6rem;

    span {
      font-weight: bold;
    }

    ${(p: IWarnigWrapper) =>
      p.color === 'white' &&
      css`
        border: 1px solid ${theme.colors.white};
        color: ${theme.colors.white};
      `}
  `}
`;
