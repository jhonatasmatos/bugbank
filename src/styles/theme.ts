export const theme = {
  colors: {
    primary: '#D62A92',
    secondary: '#A422E3',
    tertiary: '#E224ED',
    gray: '#d3d3d3',
    black: '#171717',
    white: '#fff',
  },
  font: {
    family: {
      default: "'Nunito', sans-serif",
    },
  },
  sizes: {
    xsmall: '8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
    xhuge: '6.4rem',
  },
  media: {
    lteMedium: '(max-width: 768px)',
  },
  spacings: {
    xsmall: '8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
    xhuge: '6.4rem',
  },
};
