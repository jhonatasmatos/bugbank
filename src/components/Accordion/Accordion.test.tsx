import { screen, fireEvent } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { Accordion } from '.';

const fn = jest.fn();

describe('<Accordion />', () => {
  it('should render accordion closed when tittle and accordionOpened are different', () => {
    renderTheme(
      <Accordion
        id="xpto-1234-0000"
        onClick={fn}
        accordionOpened="xpto-1234-0000"
        title="xpto-12345-0000"
        description={['Description 1', 'Description 2']}
      />,
    );

    const iconPlus = screen.queryByTestId('plus-icon') as HTMLImageElement;
    const accordionBody = screen.queryByTestId('textAccordion-xpto-1234-0000');

    expect(accordionBody).not.toBeInTheDocument();
    expect(iconPlus).toBeVisible();
  });

  it('should render accordion opened when tittle and accordionOpened are equals', () => {
    renderTheme(
      <Accordion
        id="xpto-1234-0000"
        onClick={fn}
        accordionOpened="xpto-1234-0000"
        title="xpto-1234-0000"
        description={['Description 1', 'Description 2']}
      />,
    );

    const iconMinus = screen.queryByTestId('minus-icon') as HTMLImageElement;
    const accordionBody = screen.queryByTestId('textAccordion-xpto-1234-0000');

    expect(accordionBody).toBeInTheDocument();
    expect(iconMinus).toBeVisible();
  });

  it('should click once time', () => {
    const { getByText } = renderTheme(
      <Accordion
        id="xpto-1234-0000"
        onClick={fn}
        accordionOpened="xpto-1234-0000"
        title="xpto-1234-0000"
        description={['Description 1', 'Description 2']}
      />,
    );

    const accordionHeader = getByText('xpto-1234-0000');
    fireEvent.click(accordionHeader);

    expect(fn).toBeCalledTimes(1);
  });

  it('should render with accordion closed', () => {
    const { container } = renderTheme(
      <Accordion
        id="xpto-1234-0000"
        onClick={fn}
        accordionOpened="xpto-1234-0000"
        title="xpto-12345-0000"
        description={['Description 1', 'Description 2']}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
