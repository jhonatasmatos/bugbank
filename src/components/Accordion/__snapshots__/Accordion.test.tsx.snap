// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> should render with accordion closed 1`] = `
.c0 {
  width: 100%;
  padding: 0 8rem;
  margin-bottom: 1.6rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 6rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0 1.6rem;
  cursor: pointer;
  position: relative;
  color: #fff;
  border: 1px solid #fff;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      id="accordion-xpto-1234-0000"
    >
      xpto-12345-0000
      <svg
        data-testid="plus-icon"
        fill="none"
        height="26"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="26"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="12"
          x2="12"
          y1="5"
          y2="19"
        />
        <line
          x1="5"
          x2="19"
          y1="12"
          y2="12"
        />
      </svg>
    </div>
  </div>
</div>
`;
